core/src/main/java/com/linecorp/armeria/client/Backoff.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client;
+
+import static com.linecorp.armeria.client.Backoff.FixedBackoff.NO_DELAY;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.function.LongSupplier;
+
+import com.google.common.collect.ImmutableList;
+
+/**
+ * Control back off between attemtps in a single retry operation.
+ */
+@FunctionalInterface
+public interface Backoff {
+    /**
+     * Returns a new back off value in milliseconds.
+     */
+    long nextIntervalMillis();
+
+    /**
+     * Returns a {@link Backoff} that provides zero interval.
+     */
+    static Backoff withoutDelay() {
+        return NO_DELAY;
+    }
+
+    /**
+     * Returns a {@link Backoff} that provides a fixed interval between two attempts.
+     */
+    static Backoff fixed(long intervalMillis) {
+        return new FixedBackoff(intervalMillis);
+    }
+
+    /**
+     * Returns a {@link Backoff} that provides an interval (that increases exponentially) between two attempts.
+     */
+    static Backoff exponential(long minIntervalMillis, long maxIntervalMillis) {
```

Comment:
How about adding a variant that accepts a multipler? i.e.

```java
static Backoff exponential(long, long) {
    return exponential(..., 2.0f);
}

static Backoff exponential(long, long, float multiplier) {
    ...
}
```

Perhaps you will want to implement this in `ExponentialBackoff` with an additional constructor.